pkg/collector/corechecks/network-devices/cisco-sdwan/report/metadata.go

```diff
@@ -0,0 +1,32 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package report
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform"
+	devicemetadata "github.com/DataDog/datadog-agent/pkg/networkdevice/metadata"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// TimeNow useful for mocking
+var TimeNow = time.Now
+
+// SendMetadata send Cisco SD-WAN device, interface and IP Address metadata
+func (ms *SDWanSender) SendMetadata(devices []devicemetadata.DeviceMetadata, interfaces []devicemetadata.InterfaceMetadata, ipAddresses []devicemetadata.IPAddressMetadata) {
+	collectionTime := TimeNow()
+	metadataPayloads := devicemetadata.BatchPayloads(ms.namespace, "", collectionTime, devicemetadata.PayloadMetadataBatchSize, devices, interfaces, ipAddresses, nil, nil, nil)
+	for _, payload := range metadataPayloads {
+		payloadBytes, err := json.Marshal(payload)
+		if err != nil {
+			log.Warnf("Error marshalling Cisco SD-WAN metadata : %s", err)
```

Comment:
Is this the same level we'd log at for a similar error in SNMPland? Since we're sending NDM metadata anyway, could we use the SNMP metadata sender and make it more generic?

Reply:
Changed this to an error in caf0482, good catch thanks!
We share the batching logic with the SNMP metadata sender, for the actual metadata sending though ([report_device_metadata.go](https://github.com/DataDog/datadog-agent/blob/06715aeb0c014beda7d0773547e10a584f65e7e3/pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go#L53)) it holds a lot of SNMP specific logic 😕